features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt

```diff
@@ -193,30 +227,52 @@ internal class SessionReplayFeature(
                 InternalLogger.Target.USER,
                 { EVENT_MISSING_MANDATORY_FIELDS }
             )
-            return
+            return null
         }
 
-        if (currentRumSessionId.get() == sessionId) {
-            // we already handled this session
-            return
+        return SessionData(keepSession, sessionId)
+    }
+
+    @Suppress("ReturnCount")
+    private fun shouldHandleEvent(sessionData: SessionData): Boolean {
```

Comment:
this logic is too complex and I think it needs a bit of refactor. We need to split the responsibilities here. This function is currently doing the following:"
- checking if the event makes sense to be handled 
- apply the new sampling for the new session
- switching the `shouldRecord` to true

I think based on these responsibilities we can create 3 functions here and make the things more clear.

Reply:
👍  I've split this method up